client_test.go

```diff
@@ -0,0 +1,83 @@
+package client
+
+import (
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+
+func TestNewClient(t *testing.T) {
+	Convey("Given acceptable input values", t, func(){
+		c, err := NewClient("https://localhost:4646", "testdata/testCertFile", false)
+		Convey("When a newClient is created", func() {
+			Convey("Then a client is returned with no errors", func(){
+				So(c, ShouldNotBeNil)
+				So(err, ShouldBeNil)
+			})
+		})
+	})
+}
+
+func TestNewClientFails(t *testing.T) {
+	Convey("Given invalid input values", t, func(){
+		c, err := NewClient("https://localhost:4646", "", false)
+		Convey("When a new Client is created", func(){
+			Convey("Then no client is returned with errors", func(){
+				So(err, ShouldNotBeNil)
+				So(c, ShouldBeNil)
+				So(err.Error(), ShouldStartWith, "invalid configuration with https")
+			})
+		})
+	})
+}
+
+func TestNewClientHTTP(t *testing.T) {
+	Convey("Given valid input values", t, func(){
+		c, err := NewClient("http://localhost:4646", "", false)
+		Convey("When a new Client is created", func(){
+			Convey("Then a client is returned with no errors", func(){
+				So(err, ShouldBeNil)
+				So(c, ShouldNotBeNil)
+			})
+		})
+	})
+}
+
+func TestCreateTLSConfigCert(t *testing.T) {
+	Convey("Given valid input values", t, func(){
```

Comment:
"Given valid input values" is correct but doesn't give much context.  Something like:

`Given nomadCACert is empty and nomadTLSSkipVerify is true`

provides a clearer outline of scenario being test. I know this sounds like _nit-picky_ point but it's helpful to anyone looking at this in the future. If they need to update this for some reason being able quickly find the test they need makes their life that little bit easier.